sdk/eventhub/Azure.Messaging.EventHubs/src/Core/BasicRetryPolicy.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Globalization;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Errors;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   The default retry policy for the Event Hubs client library, respecting the
+    ///   configuration specified as a set of <see cref="RetryOptions" />.
+    /// </summary>
+    ///
+    /// <seealso cref="RetryOptions"/>
+    ///
+    internal class BasicRetryPolicy : EventHubRetryPolicy
+    {
+        /// <summary>The seed to use for initializing random number generated for a given thread-specific instance.</summary>
+        private static int s_randomSeed = Environment.TickCount;
+
+        /// <summary>The random number generator to use for a specific thread.</summary>
+        private static readonly ThreadLocal<Random> s_random = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref s_randomSeed)), false);
+
+        /// <summary>
+        ///   The set of options responsible for configuring the retry
+        ///   behavior.
+        /// </summary>
+        ///
+        public RetryOptions Options { get; }
+
+        /// <summary>
+        ///   The factor to apply to the base delay for use as a base jitter value.
+        /// </summary>
+        ///
+        /// <value>This factor is used as the basis for random jitter to apply to the calculated retry duration.</value>
+        ///
+        public double JitterFactor { get; } = 0.08;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="BasicRetryPolicy"/> class.
+        /// </summary>
+        ///
+        /// <param name="retryOptions">The options which control the retry approach.</param>
+        ///
+        public BasicRetryPolicy(RetryOptions retryOptions)
+        {
+            Guard.ArgumentNotNull(nameof(retryOptions), retryOptions);
```

Comment:
You're passing in  the name and the value so that any resulting `ArgumentException` gets a representative name from the calling context rather than the name of the param used by `Guard`.    The value is what is checked against.